app/models/katello/rhsm_fact_parser.rb

```diff
@@ -54,11 +54,17 @@ def operatingsystem
       os_name = ::Katello::Candlepin::Consumer.distribution_to_puppet_os(name)
       major, minor = version.split('.')
       if os_name && !invalid_centos_os?(os_name, minor)
-        os_attributes = {:major => major, :minor => minor || '', :name => os_name}
+        os_attributes = {:major => major, :minor => minor || '', :name => os_name, :release_name => os_release_name(os_name)}
         ::Operatingsystem.find_by(os_attributes) || ::Operatingsystem.create!(os_attributes)
       end
     end
 
+    def os_release_name(os_name)
+      if os_name == 'Debian'
+        facts['distribution.id']
+      end
```

Comment:
Just a thought. This might be needed for Ubuntu, too. Can we use os_family?

Reply:
Do you mean you want to set the `release_name` and `family` on the OS?

Reply:
No, i meant to set the `release_name` in case the os_family is Debian, which includes Ubuntu (and probably > 100 more not yet known to foreman).

Reply:
Done!